packages/dev-kit/src/rpc/wrap-rpc.ts

```diff
@@ -0,0 +1,126 @@
+import type { ChildProcess } from 'child_process';
+
+import { RpcExitError } from './rpc-error';
+import { RpcRemoteMethod, RpcMessage, RpcGMCallTypes } from './types';
+
+interface WrapRpcOptions {
+  id: string;
+  once?: boolean;
+}
+
+function createControlledPromise<T = unknown>() {
+  let resolve: (value: T) => void = () => undefined;
+  let reject: (error: unknown) => void = () => undefined;
+  const promise = new Promise<T>((aResolve, aReject) => {
+    resolve = aResolve;
+    reject = aReject;
+  });
+
+  return {
+    promise,
+    resolve,
+    reject,
+  };
```

Comment:
instead of this, you could try `Promise.withResolvers`

Reply:
really new feature , but we need to support node@16 , so can not use this